ciso/_ciso.pyx

```diff
@@ -40,14 +40,15 @@ def zslice(double[:, :, ::1] q,
 
     cdef double[:, ::1] q_iso = np.empty((M, L), dtype=np.float64)
 
-    for i in range(L):
-        for j in range(M):
-            q_iso[j, i] = mask_val
-            for k in range(N-1):
-                if (((p[k, j, i] < p0) and (p[k+1, j, i] > p0)) or
-                   ((p[k, j, i] > p0) and (p[k+1, j, i] < p0))):
-                    dp = p[k+1, j, i] - p[k, j, i]
-                    dp0 = p0 - p[k, j, i]
-                    dq = q[k+1, j, i] - q[k, j, i]
-                    q_iso[j, i] = q[k, j, i] + dq*dp0/dp
+    with nogil:
```

Comment:
As far as I know, the nogil will only help here.

But in theory, some other thread could mangle the q and p arrays while this code is in the middle of the loop. Coudl that happen with our use cases???

-CHB